coordinator/src/node/expired_positions.rs

```diff
@@ -33,70 +36,88 @@ pub async fn close(node: Node) {
+        )? {
+            let trader_id = order.trader_id.to_string();
+            let order_id = order.id.to_string();
 
-        if !node.is_connected(&position.trader) {
-            tracing::debug!(
-                "Could not close expired position with {} as trader is not connected.",
-                position.trader
-            );
-            continue;
-        }
+            if order.expiry < OffsetDateTime::now_utc() {
+                tracing::warn!(trader_id, order_id, "Matched order expired! Giving up on that position, looks like the corresponding dlc channel has to get force closed.");
+                orderbook::db::orders::set_order_state(&mut conn, order.id, OrderState::Failed)?;
+
+                orderbook::db::matches::set_match_state_by_order_id(
+                    &mut conn,
+                    order.id,
+                    MatchState::Failed,
+                )?;
 
-        let channel_id = match node.decide_trade_action(&position.trader) {
-            Ok(TradeAction::Close(channel_id)) => channel_id,
-            Ok(_) => {
-                tracing::error!(?position, "Unable to find sub channel of expired position.");
+                let amounts = orderbook::db::matches::get_matches_by_order_id(&mut conn, order.id)?
+                    .iter()
+                    .map(|m| QuantityPrice {
+                        quantity: m.quantity,
+                        execution_price: m.execution_price,
+                    })
+                    .collect();
+
+                let closing_price = average_execution_price(amounts)
+                    .to_f32()
+                    .expect("to fit into f32");
+                db::positions::Position::set_open_position_to_closing(
+                    &mut conn,
+                    position.trader.to_string(),
+                    closing_price,
+                )?;
                 continue;
-            }
-            Err(e) => {
-                tracing::error!(?position, "Failed to decide trade action. Error: {e:?}");
+            } else {
+                tracing::trace!(trader_id, order_id, "Skipping expired position as match has already been found. Waiting for trader to come online to execute the trade.");
                 continue;
             }
-        };
+        }
 
-        let closing_price =
-            match BitmexClient::get_quote(&node.inner.network, &position.expiry_timestamp).await {
-                Ok(quote) => quote.get_price_for_direction(position.direction.opposite()),
-                Err(e) => {
-                    tracing::warn!(
-                        "Failed to get quote from bitmex for {} at {}. Error: {e:?}",
-                        position.trader,
-                        position.expiry_timestamp
-                    );
-                    continue;
-                }
-            };
+        tracing::debug!(trader_pk=%position.trader, %position.expiry_timestamp, "Attempting to close expired position");
```

Comment:
is `%position.trader` printed nicely?

Reply:
> The % sigil operates similarly, but indicates that the value should be recorded using its [fmt::Display](https://doc.rust-lang.org/nightly/core/fmt/trait.Display.html) implementation:

https://docs.rs/tracing/latest/tracing/

Reply:
That's not helpful 😅 will it be printed as `8b143` or as `[08,11,20,03,...` ? The latter one is not really helpful and we have heaps of them already, hence I'm asking.